Sources/AWSLambdaRuntimeCore/LambdaContext.swift

```diff
@@ -49,40 +50,118 @@ extension Lambda {
 extension Lambda {
     /// Lambda runtime context.
     /// The Lambda runtime generates and passes the `Context` to the Lambda handler as an argument.
-    public final class Context: CustomDebugStringConvertible {
+    public struct Context: BaggageContext.Context, CustomDebugStringConvertible {
+        private var storage: _Storage
+
+        final class _Storage {
+            var baggage: Baggage
+
+            let invokedFunctionARN: String
+            let deadline: DispatchWallTime
+            let cognitoIdentity: String?
+            let clientContext: String?
```

Comment:
technically those could also become baggage items, your call really folks.

Only the trace stuff matters for instruments/tracers/logging, which is why I put it there.

Reply:
If we have those values are present, we should add them to the baggage. Will associate the trace with a user. Though this could raise problems with privacy. Make this configurable?

Reply:
I mean... client context is pretty special, it's actually JSON:

![Screen Shot 2020-09-16 at 19 23 14](https://user-images.githubusercontent.com/120979/93325301-11ed6a00-f852-11ea-9f24-44b8d6e7778f.png)

with some known magic values or "custom".

I don't think it's necessary in the baggage tbh...? Those values dont seem like they are important for tracing but just for this specific node / lambda?

https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html

Should the ClientContext be modeled using proper types?

Reply:
Oh boy, reading into the docu we should definitely fix that. But I think we should target another PR. I'm afraid though that it will be really hard to wrap our head around the `Custom` value here. I guess we will need something like `AnyCodable` for this.

Reply:
I mean, custom is simply a String... :)